src/csv_overrides.py

```diff
@@ -0,0 +1,141 @@
+from talon import Context, actions, fs, app
+import os
+from datetime import datetime
+from pathlib import Path
+
+directory_name = "cursorless-settings"
+
+ctx = Context()
+
+
+def init_csv_and_watch_changes(filename: str, default_values: dict[str, dict]):
+    dir_path, file_path = get_file_paths(filename)
+    super_default_values = get_super_values(default_values)
+
+    if not dir_path.is_dir():
+        os.mkdir(dir_path)
+
+    def on_watch(path, flags):
+        if file_path.match(path):
+            current_values = read_file(file_path, super_default_values.values())
+            update_dicts(default_values, current_values)
+
+    fs.watch(dir_path, on_watch)
+
+    if file_path.is_file():
+        current_values = update_file(file_path, super_default_values)
+        update_dicts(default_values, current_values)
+    else:
+        create_file(file_path, super_default_values)
+        update_dicts(default_values, super_default_values)
+
+
+def update_dicts(default_values: dict[str, dict], current_values: dict):
+    # Create map with all default values
+    results_map = {}
+    for list_name, dict in default_values.items():
+        for key, value in dict.items():
+            results_map[value] = {"key": key, "value": value, "list": list_name}
+
+    # Update result with current values
+    for key, value in current_values.items():
+        results_map[value]["key"] = key
+
+    # Convert result map back to result list
+    results = {key: {} for key in default_values}
+    for obj in results_map.values():
+        results[obj["list"]][obj["key"]] = obj["value"]
+
+    # Assign result to talon context list
+    for list_name, dict in results.items():
+        ctx.lists[f"user.cursorless_{list_name}"] = dict
```

Comment:
Hmm this one is a bit too magical for me.  I'd be inclined to just have the list name be the full talon list name.

Reply:
We used this kind of logic in more places. I actually think it's nice that we force this prefix and don't allow any other format.

Reply:
Should we move the list name construction to a utility function then?  Happy to do it

Reply:
Yeah do that.